src/LEGO.AsyncAPI.Bindings/Sqs/IPrincipalValue.cs

```diff
@@ -0,0 +1,54 @@
+// Copyright (c) The LEGO Group. All rights reserved.
+
+namespace LEGO.AsyncAPI.Bindings.Sqs;
+
+using System;
+using System.Collections.Generic;
+using LEGO.AsyncAPI.Writers;
+
+public interface IPrincipalValue
```

Comment:
Consider adding a "T GetValue()" method to this interface.
Once the objects has been deserialized when reading.
We might want to get their values.

GetValue would be a nice method to have instead of exposing the inner values directly as properties.